### ISSUE TITLE
Error when try to change AWS credentials "TypeError: Cannot read property 'type' of undefined"

### ISSUE THREAD

**Issue by sytolk** [aws-amplify/amplify-cli#6635]
**Amplify CLI Version**
amplify -v
4.42.0

**To Reproduce**
Steps to reproduce :
Amplify Project is working and published. I have try to change AWS profile in order to publish in other AWS account

`amplify configure project`

```
> amplify configure project

? Enter a name for the project tagspaces
? Choose your default editor: IntelliJ IDEA
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using react
? Source Directory Path:  tagspaces/app
? Distribution Directory Path: tagspaces/web
? Build Command:  npm run-script build
? Start Command: npm run-script start
Using default provider  awscloudformation
? Do you want to enable container-based deployments? No

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

For the awscloudformation provider.
? Do you want to update or remove the project level AWS profile? Update AWS Profile
Error occurred during configuration.
TypeError: Cannot read property 'type' of undefined
    at promptForAuthConfig (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:349:33)
    at update (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:262:11)
    at carryOutConfigAction (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:189:22)
    at Object.configure (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:86:16)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tagspaces-amplify@1.0.0 configure: `amplify configure project`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tagspaces-amplify@1.0.0 configure script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sytolk/.npm/_logs/2021-02-16T07_43_49_415Z-debug.log

Process finished with exit code 1
```

**Log output**
Include any relevant log output under `~/.amplify/logs/amplify-cli-<issue-date>.log`
[2021-02-16T07_43_49_415Z-debug.log](https://github.com/aws-amplify/amplify-cli/files/5986932/2021-02-16T07_43_49_415Z-debug.log)

**Comment by sytolk** [comment]
I have try to change profileName in local-aws-info.json 
```
{
  "dev": {
    "configLevel": "project",
    "useProfile": true,
    "profileName": "awsclient"
  }
}
```
but after `amplify status`
```
 Fetching AWS Amplify Console domainsProfile configuration is missing for: awsclient
Error: Profile configuration is missing for: awsclient
    at Object.getProfiledAwsConfig (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/system-config-manager.js:90:17)
    at loadConfigurationForEnv (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:568:45)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Object.loadConfiguration (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:524:18)
    at Object.aws.configureWithCreds (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/aws-utils/aws.js:30:18)
    at Object.getConfiguredAWSClient (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/index.ts:67:3)
    at getAWSClient (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-console-hosting/utils/client-factory.js:16:10)
    at Object.getAmplifyClient (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-console-hosting/utils/client-factory.js:4:15)
    at Object.generateTableContentForApp (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-console-hosting/utils/table-utils.js:8:25)
    at Object.status (/Users/sytolk/.nvm/versions/node/v12.13.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-console-hosting/hosting/index.js:139:3)

```

**Comment by sharmaaditya570191** [comment]
I am able to reproduce this error on the following configuration:-

Node version: 10.23.3
OS: Ubuntu 18.04
Amplify CLI version: 4.43.0

I used `amplify configure project` and got this error.
![image](https://user-images.githubusercontent.com/42587159/108105083-27d9b200-70b2-11eb-86f5-f1e56bf16ffd.png)

**Comment by renebrandel** [comment]
@sharmaaditya570191 @sytolk thanks for raising this issue. #6629 will address this problem and we're going to release it ASAP.

**Comment by jhockett** [comment]
A fix was released in version 4.44.0. If you run into this issue again, please comment below.
